pallets/author-mapping/src/tests.rs

```diff
@@ -0,0 +1,279 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Unit testing
+use crate::mock::{
+	last_event, AuthorMapping, Balances, Event as MetaEvent, ExtBuilder, Origin, System, Test,
+	TestAuthor,
+};
+use crate::{Error, Event};
+use frame_support::{assert_noop, assert_ok};
+
+#[test]
+fn genesis_builder_works() {
+	ExtBuilder::default()
+		.with_balances(vec![(1, 1000)])
+		.with_mappings(vec![(TestAuthor::Alice, 1)])
+		.build()
+		.execute_with(|| {
+			assert!(System::events().is_empty());
+			assert_eq!(Balances::free_balance(&1), 900);
+			assert_eq!(Balances::reserved_balance(&1), 100);
+			assert_eq!(AuthorMapping::account_id_of(TestAuthor::Alice), Some(1));
```

Comment:
Where can I find the address for TestAuthor::Alice ? Where is the genesis default config btw? thanks in advance

Reply:
This is a mock runtime meant for unit tests only. Unlike the real runtime which has `type AuthorId = NimbusId`, here I've just used this simple enum `type AuthorId = TestAuthorId`. So there is no 0x123 looking thing associated with `TestAuthor::Alice`. Rather the AuthorId _is_ `TestAuthor::Alice`.